src/foam/core/Listener.js

```diff
@@ -113,6 +118,17 @@ foam.CLASS({
         configurable: true,
         enumerable: false
       });
+
+      if ( on.length > 0 ) {
+        var listener = proto[this.name];
+        for ( var i = 0 ; i < on.length ; i++ ) {
+          let o        = on[i].split('.');
+          let objectOn = o.shift();
+          let topic    = o;
+          console.log(listener, topic);
+          proto.onDetach(proto.sub.apply(proto, topic.concat(listener)));
```

Comment:
We don't need an onDetach() when the listener is on that object anyway, because when proto is detached it will automatically detach all listeners listening to it. Only needed when listening to a different object.